lib/dependency_validator.dart

```diff
@@ -170,9 +170,15 @@ void run({List<String> ignoredPackages = const []}) {
           // Remove all deps being used for their transformer(s)
           .difference(packagesUsedViaTransformers)
             // Remove this package, since we know they're using our executable
-            ..remove(dependencyValidatorPackageName)
-            // Ignore known unused packages
-            ..removeAll(ignoredPackages);
+            ..remove(dependencyValidatorPackageName);
+
+  if (unusedDependencies.contains('analyzer')) {
+    logger.warning('You do not need to depend on the analyzer to run the dart analyzer.');
```

Comment:
#nit wording is a little off in `on the analyzer`, `Dart` is not capitalized

Might also be nice to say what you should do instead.
```
You do not need to depend on `analyzer` to run the Dart analyzer.
Instead, just run the `dartanalyzer` executable that is bundled with the Dart SDK.
```